### PR TITLE
feat(ironfish): Use `assetIdentifier` when saving unconfirmed balance

### PR DESCRIPTION
**Pull request by rohanjadvani**
## Summary

Pass in asset identifiers from notes when updating balances.

## Testing Plan

N/A

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish-cli/src/commands/accounts/repair.ts [hunk 1]
```diff
@@ -109,7 +110,8 @@ export default class Repair extends IronfishCommand {
     walletDb: WalletDB,
     chain: Blockchain,
   ): Promise<void> {
-    let unconfirmedBalance = 0n
+    const unconfirmedBalances = new BufferMap<bigint>()
+    const unconfirmedBalance = 0n
```
Comment (mat-if) [inline ironfish-cli/src/commands/accounts/repair.ts:114]: I don't think we need this variable anymore - it's only used for the log, which probably needs to be updated too

File: ironfish-cli/src/commands/accounts/repair.ts [hunk 2]
```diff
@@ -156,15 +158,23 @@ export default class Repair extends IronfishCommand {
...
+          const unconfirmedBalance = unconfirmedBalances.get(assetIdentifier) ?? BigInt(0)
+          unconfirmedBalances.set(
+            assetIdentifier,
+            unconfirmedBalance + decryptedNoteValue.note.value(),
+          )
         }
       }
     }
 
     this.log(
```
Comment (mat-if) [inline ironfish-cli/src/commands/accounts/repair.ts:171]: We should move this into the loop below to output for each asset that it updates

**Review by hughy — APPROVED** [review]

**Review by mat-if — APPROVED** [review]
Should we add some test coverage for this? We get a lot for free for the native asset, but just to confirm it's working for a custom asset as well

File: ironfish-cli/src/commands/accounts/repair.ts (hunk 2 above)
Comment (rohanjadvani) [reply ironfish-cli/src/commands/accounts/repair.ts]: Done

File: ironfish-cli/src/commands/accounts/repair.ts (hunk 1 above)
Comment (rohanjadvani) [reply ironfish-cli/src/commands/accounts/repair.ts]: Ah yeah forgot to clean it up, thanks

**Pull request merged**